var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -99,6 +99,8 @@ class Petsc(Package):
             description='Activate X support')
     variant('batch', default=False,
             description='Enable when mpiexec is not available to run binaries')
+    variant('valgrind', default=False,
+            description='Activates support for Valgrind')
```

Comment:
I would change this to: "Enable Valgrind Client Request mechanism"

https://valgrind.org/docs/manual/manual-core-adv.html#manual-core-adv.clientreq

We should improve this documentation in petsc

Reply:
Done

Reply:
Note: "Activates support for Valgrind" might suggest - petsc codes can't be debugged with valgrind - if  build with default  petsc~valgind.  This is incorrect.

petsc+valgrind basically enables it to check if the code is run under valgrind - if so - it can disable its internal (memory) checks in favor of valgrind's checks.